fwprovider/acme/datasource_acme_plugins.go

```diff
@@ -0,0 +1,171 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+package acme
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/int64validator"
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/bpg/terraform-provider-proxmox/proxmox"
+	"github.com/bpg/terraform-provider-proxmox/proxmox/cluster/acme/plugins"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &acmePluginsDatasource{}
+	_ datasource.DataSourceWithConfigure = &acmePluginsDatasource{}
+)
+
+// NewACMEPluginsDataSource is a helper function to simplify the provider implementation.
+func NewACMEPluginsDataSource() datasource.DataSource {
+	return &acmePluginsDatasource{}
+}
+
+// acmePluginsDatasource is the data source implementation for ACME plugins.
+type acmePluginsDatasource struct {
+	client *plugins.Client
+}
+
+// acmePluginsModel maps the schema data for the ACME plugins data source.
+type acmePluginsModel struct {
+	Plugins []acmePluginModel `tfsdk:"plugins"`
+}
+
+// acmePluginModel maps the schema data for an ACME plugin.
+type acmePluginModel struct {
+	API             types.String `tfsdk:"api"`
+	Data            types.Map    `tfsdk:"data"`
+	Digest          types.String `tfsdk:"digest"`
+	Plugin          types.String `tfsdk:"plugin"`
+	Type            types.String `tfsdk:"type"`
+	ValidationDelay types.Int64  `tfsdk:"validation_delay"`
+}
+
+// Metadata returns the data source type name.
+func (d *acmePluginsDatasource) Metadata(
+	_ context.Context,
+	req datasource.MetadataRequest,
+	resp *datasource.MetadataResponse,
+) {
+	resp.TypeName = req.ProviderTypeName + "_acme_plugins"
+}
+
+// Schema returns the schema for the data source.
+func (d *acmePluginsDatasource) Schema(
+	_ context.Context,
+	_ datasource.SchemaRequest,
+	resp *datasource.SchemaResponse,
+) {
+	resp.Schema = schema.Schema{
+		Description: "Retrieves the list of ACME plugins.",
+		Attributes: map[string]schema.Attribute{
+			"plugins": schema.ListNestedAttribute{
```

Comment:
The API docs describe this as a list of plugin names and `pvesh` returns just a list of names, but actually the API returns a list of objects: https://pve.proxmox.com/pve-docs/api-viewer/#/cluster/acme/plugins

Should we follow the documentation and `pvesh` and just retrieve a list of strings or use what we get from the API and retrieve the list of objects?